MediaBrowser.Model/Dlna/CodecProfile.cs

```diff
@@ -1,74 +1,94 @@
-            }
+    /// <summary>
+    /// Gets or sets the subcontainer(s) which this profile will be applied to.
+    /// </summary>
+    [XmlAttribute("subcontainer")]
+    public string? SubContainer { get; set; }
 
-            var codecs = GetCodecs();
-            if (codecs.Length == 0)
-            {
-                return true;
-            }
+    /// <summary>
+    /// Checks to see whether the codecs and containers contain the given parameters.
+    /// </summary>
+    /// <param name="codecs">The codecs to match.</param>
+    /// <param name="container">The container to match.</param>
+    /// <param name="useSubContainer">Consider subcontainers.</param>
+    /// <returns>True if both conditions are met.</returns>
+    public bool ContainsAnyCodec(string[] codecs, string? container, bool useSubContainer = false)
+    {
+        var containerToCheck = useSubContainer && string.Equals(Container, "hls", StringComparison.OrdinalIgnoreCase) ? SubContainer : Container;
+        return ContainerHelper.ContainsContainer(containerToCheck, container) && codecs.Any(c => ContainerHelper.ContainsContainer(Codec, false, c));
+    }
 
-            foreach (var val in codec)
-            {
-                if (codecs.Contains(val, StringComparison.OrdinalIgnoreCase))
-                {
-                    return true;
-                }
-            }
+    /// <summary>
+    /// Checks to see whether the codecs and containers contain the given parameters.
+    /// </summary>
+    /// <param name="codec">The codec to match.</param>
+    /// <param name="container">The container to match.</param>
+    /// <param name="useSubContainer">Consider subcontainers.</param>
+    /// <returns>True if both conditions are met.</returns>
+    public bool ContainsAnyCodec(string? codec, string? container, bool useSubContainer = false)
+    {
+        var containerToCheck = useSubContainer && string.Equals(Container, "hls", StringComparison.OrdinalIgnoreCase) ? SubContainer : Container;
+        return ContainerHelper.ContainsContainer(containerToCheck, container) && ContainerHelper.ContainsContainer(Codec, false, codec);
+    }
 
-            return false;
-        }
+    /// <summary>
+    /// Checks to see whether the codecs and containers contain the given parameters.
+    /// </summary>
+    /// <param name="codec">The codec to match.</param>
+    /// <param name="container">The container to match.</param>
+    /// <param name="useSubContainer">Consider subcontainers.</param>
+    /// <returns>True if both conditions are met.</returns>
+    public bool ContainsAnyCodec(ReadOnlySpan<char> codec, string? container, bool useSubContainer = false)
+    {
+        var containerToCheck = useSubContainer && string.Equals(Container, "hls", StringComparison.OrdinalIgnoreCase) ? SubContainer : Container;
+        return ContainerHelper.ContainsContainer(containerToCheck, false, container) && ContainerHelper.ContainsContainer(Codec, false, codec);
```

Comment:
It looks like `Container` supported inverse matching.
```suggestion
        return ContainerHelper.ContainsContainer(containerToCheck, container) && ContainerHelper.ContainsContainer(Codec, false, codec);
```